include/coap/option.h

```diff
@@ -407,4 +409,68 @@ coap_opt_const_value( const coap_opt_t *opt ) {
 
 /** @} */
 
+/**
+ * Representation of chained list of CoAP options to install.
+ *
+ * @code
+ * coap_optlist_t *optlist_head = NULL;
+ * coap_pdu_t *pdu = coap_new_pdu(session);
+ *
+ * ... other set up code ...
+ * coap_insert_optlist(&optlist, coap_new_optlist(COAP_OPTION_OBSERVE,
+ *                    COAP_OBSERVE_ESTABLISH, NULL));
+ *
+ * coap_add_optlist_pdu(pdu, *optlist_head);
+ * @endcode
+ */
+typedef struct coap_optlist_t {
+  struct coap_optlist_t *next;  /**< next entry in the optlist chain */
+  uint16_t number;              /**< the option number (no delta coding) */
+  unsigned int length;          /**< the option value length */
+  uint8_t data[];               /**< the option data */
+} coap_optlist_t;
+
+/**
+ * Create a new optlist entry.
+ *
+ * @param number    The option number (COAP_OPTION_*)
+ * @param length    The option length
+ * @param data      The option value data
+ *
+ * @return          A pointer to the new optlist entry, or @c NULL if error
+ */
+coap_optlist_t *coap_new_optlist(uint16_t number,
+                                 size_t length,
+                                 const uint8_t *data);
+
+/**
+ * The current optlist of @p chain is first sorted (as per RFC7272 ordering
+ * requirements) and then added to the @pdu.
+ *
+ * @param pdu  The pdu to set the options on from the chain list
+ * @param optlist_chain The chained list of optlist to add to the pdu
+ *
+ * @return     @c 1 if succesful or @c 0 if failure;
+ */
+int coap_add_optlist_pdu(coap_pdu_t *pdu, coap_optlist_t* optlist_chain);
+
+/**
+ * Adds @p optlist to the given @p optlist_chain which must initially
```

Comment:
"which must initially be set to NULL" is confusing. What must be NULL? When?

And who is the owner of the elements in `optlist` after it has been added to `*optlist_chain`?